Code/Peng_Robinson_Class.m

```diff
@@ -0,0 +1,283 @@
+                 
+                 
+                 Z(L_Case2,1) = (u(L_Case2)+v(L_Case2))-a_card(L_Case2)/3;
+                 Z(L_Case2,2) = (u(L_Case2)*epsilon_1 + v(L_Case2)*epsilon_2)-a_card(L_Case2)/3;
+                 Z(L_Case2,3) = (u(L_Case2)*epsilon_2 + v(L_Case2)*epsilon_1)-a_card(L_Case2)/3;
+                 
+                 Z(L_Case1,1) = ( sqrt(-(4/3).*p_card(L_Case1)).*cos((1/3).*acos(-(q_card(L_Case1)./2).*sqrt(-27./p_card(L_Case1).^3))))       - a_card(L_Case1)./3; 
+                 Z(L_Case1,2) = (-sqrt(-(4/3).*p_card(L_Case1)).*cos((1/3).*acos(-(q_card(L_Case1)./2).*sqrt(-27./p_card(L_Case1).^3))+pi./3)) - a_card(L_Case1)./3;
+                 Z(L_Case1,3) = (-sqrt(-(4/3).*p_card(L_Case1)).*cos((1/3).*acos(-(q_card(L_Case1)./2).*sqrt(-27./p_card(L_Case1).^3))-pi./3)) - a_card(L_Case1)./3;
+                                  
+                 Vm = (T.*Z.*R)./obj.P;
+                 vm_liquid = min(Vm,[],2);
+                 vm_vapor  = max(Vm,[],2);
+                 Vm = [vm_liquid, vm_vapor];
+                                                   
+                 % Calculation of fugacities
+                 phi = @(Z) exp(Z - 1 - log(Z - B) - A./(2*2^0.5*B).*log((Z+2.414*B)./(Z-0.414*B)));
+                 
+                 Z_liquid = obj.P.*vm_liquid ./ (R*T);
+                 Z_vapor = obj.P.*vm_vapor ./ (R*T);
+                 phi_liquid = phi(Z_liquid);
+                 phi_vapor = phi(Z_vapor);
+                 
+                 converged = (abs(phi_liquid.*obj.P - phi_vapor.*obj.P)) <= 1;
+                 obj.P = real(obj.P.*phi_liquid./phi_vapor);
+                 
+                 n= n+1;
+                 if n >=100, break, end
+                 
+            end %While-Loop
+            
+            obj.Ts = obj.T(converged);
+            obj.Ps = obj.P(converged);
+            obj.Vms = Vm(converged,:);
+                                    
+        end %Pressure_Calculation
+        
+        
+%         function Temperature_Calculation(obj,P,T)
+%             %Caclulates the Saturation Temperature at a given Pressure P.
+%             %If no start Temperature is given via the input, the Temperature will
+%             %be set to the saturation Temperature calculated with the
+%             %Antoine-equation at the given Pressure. Every value in
+%             %SI-Units.
+%             %
+%             %Input        Temperature_Calculation(P,T)
+%             %
+%             % -T:         "vector"  (Start Temperature, possibility of dispense given with following if loop) 
+%             % -p:         "vector"  (Vector of pressures for which the corresponding 
+%             %                        temperatures should be calculated. Maximal from 
+%             %                        tripple point pressure to critical point pressure)
+%             %
+%             %Output
+%             %
+%             % -Ts:         "vector" (Same size as p, Calculated Saturation Temperature for 

```

Comment:
Output vectors have different size than input if not all values converge. Thats why we return Ts, the user wouldnt know which values were rejected

Reply:
Yes you are right, the Attributes T and P dont make much sense here, i will delete them soon.

But the function Calculate_Pressure should work. I tested it in coparison with the already existing PR_vectorised function. The values are nearly the same. They reason they differ is that I included two different solutions for the Cardano´s formula, depending on the sign of the discriminant. 
But the difference to the existing PR function is smaller than 0,01%.... hence not worth mentioning.

I will try to get the Calculate_Temperature function to work soon

The testing code I run is the following:

methane = Substance("methane");
T = 180:190.6;
Test1 = Peng_Robinson_Class('methane');
Test1.Pressure_Calculation(T)


[Ts, Ps, Vms] = Peng_Robinson_vectorized(methane, T);

all ( Test1.Ts == Ts)
all ( abs(Test1.Ps - Ps) < 0.0001)
all ( abs(Test1.Vms - Vms) <0.0001)

Reply:
Regarding Attributes T and P - i think if those would be the output Vectors they would actually potentially be helpful if they are stored like this - since this would make transferal of data pretty easy in comparison to how we do it atm.

So basically right now we create a textfile with the P and T data in them, and then read it out of the the textfile again. Since matlab has a pretty easy way of saving Objects and opening them (literally save and load filename) again in another file this would be way more efficient therefor i actually like the idea of doing it like this.

Reply:
Currently, we don't need to create a textfile, it is just something i used because i didn't like to invoke PR all the time i train, especially when PR was slow. We can feed our PR return values directly into our Network, theres no need to transfer anything.
My point is, that Temperature and all these Thermodynamic Data are no property of the PR EOS. I would therefore suggest to make the relevant Methods **static Methods**. The PR class would then simply be a collection of these methods, convinient due to the naming scheme, and nothing more. These methods don't need to know anything about the PR Class.
Also, when storing the thermodynamic data like that, in retrospective it is not possible to know exactly how they were generated